keras_contrib/applications/densenet.py

```diff
@@ -404,26 +425,23 @@ def __transition_up_block(ip, nb_filters, type='upsampling', output_shape=None,
     if type == 'upsampling':
         x = UpSampling2D()(ip)
     elif type == 'subpixel':
-        x = Conv2D(nb_filters, (3, 3), padding="same", kernel_regularizer=l2(weight_decay), activation='relu',
+        x = Conv2D(nb_filters, (3, 3), activation="relu", padding='same', kernel_regularizer=l2(weight_decay),
                    use_bias=False, kernel_initializer='he_uniform')(ip)
-        #x = Convolution2D(nb_filters, 3, 3, activation="relu", border_mode='same', W_regularizer=l2(weight_decay),
-        #                  bias=False, init='he_uniform')(ip)
         x = SubPixelUpscaling(scale_factor=2)(x)
         x = Conv2D(nb_filters, (3, 3), activation="relu", padding='same', kernel_regularizer=l2(weight_decay),
-                          use_bias=False, kernel_initializer='he_uniform')(x)
-    elif type == 'atrous':
-        # waiting on https://github.com/fchollet/keras/issues/4018
-        x = AtrousConvolution2D(nb_filters, 3, 3, activation="relu", W_regularizer=l2(weight_decay),
-                                bias=False, atrous_rate=(2, 2), init='he_uniform')(ip)
+                   use_bias=False, kernel_initializer='he_uniform')(x)
     else:
-        x = Deconvolution2D(nb_filters, 3, 3, output_shape, activation='relu', border_mode='same',
-                            subsample=(2, 2), init='he_uniform')(ip)
+        x = Conv2DTranspose(nb_filters, (3, 3), output_shape, activation='relu', padding='same',
+                            subsample=(2, 2), kernel_initializer='he_uniform')(ip)
 
```

Comment:
Hmm now that Keras 2 has removed the need for knowing the output shape, the code for Deconv upscaling can be heavily simplified. No more need for the batchsize parameter. Keep it for now since many people (including me) have not shifted to keras 2 yet. Once the bugs from Keras 2 are removed, we can start refactoring this repo to remove support code for keras 1 like batchsize params and output shape computation.